### ISSUE TITLE
[BUG/ENHANCE] Waypoints

### ISSUE THREAD

**Issue by FoxPotato** [DarkholmeTenk/TardisModIssues#147]
Waypoints don't actually set coordinates but only the levers, this isn't all too helpful if you've been using relative coords.

**Comment by Soyaro666** [comment]
wouldn't it be boring if the tardis could do everything on it's own?
It's pretty easy to note the coordinates you're at and set them in absolute mode. Of course it's hard on low levels since the Tardis doesn't show you the exact target coordinates but a point near it but on higher levels (level 10 I guess) it becomes very precise

**Comment by FoxPotato** [comment]
Yeah but imagine this, you're on the max of absolute coords. You want to go further, you set the coord mode to relative. then go in some random directions.
Yo set a waypoint, go to like 0,64,0. u wanna get back, but wait! you can't bc u lost it bc waypoints only go to the max of relative coords.

**Comment by DarkholmeTenk** [comment]
TARDIS only gets its controls from the levers. When you do recall, all it does is sets the levers and triggers a flight. The only exception is the key, which just places the block where you are.

**Comment by FoxPotato** [comment]
Wait, so if you recall outside of the absolute coord range can you not get picked up?

**Comment by Soyaro666** [comment]
Wait, there was an anomaly; distances in overworld are further than in the nether. 8 times further I guess. you could build a nether gate, go through it and would be at 1/8 of the distance. there you should be able to recall again. on the other way: going to maximum in Nether and building a gate to overworld would bring you 8 times that distance away. but the Tardis' reach is already a giant distance in Minecraft terms. it's just a theory and you would need to test it. maybe I'll find time to try it on my own later...

**Comment by DarkholmeTenk** [comment]
@FoxPotato yes, that's true. There's also regions in the normal coordinate range that can't be reached.

@Soyaro666 that sounds accurate.

**Comment by Soyaro666** [comment]
I could try to write a concept to close these holes but that would be a complete redo to the controls so I don't really want...
anyway. FoxPotato? I tried something out. every wheel to 7, every Lever to 6, current destination coords is x=z=395220. seems like thats the maximum a tardis can reach with absolute coordinates. fortunately you can go further by turning to relative mode later ;-)

Darkholme, could you tell me what exactly the coordinate levers and wheels do? like what's the levers value in code and how do the wheels affect them? of course that would be a lot of work for you (I don't think you have all that stuff ordered correctly in your head; or did you note them somewhere?) but it would really help me to understand the tardis much better

**Comment by FoxPotato** [comment]
"anyway. FoxPotato? I tried something out. every wheel to 7, every Lever to 6, current destination coords is x=z=395220. seems like thats the maximum a tardis can reach with absolute coordinates. fortunately you can go further by turning to relative mode later ;-)"
this is exactly what I did, like I said, I'm going to the end of the universe.

**Comment by Soyaro666** [comment]
well, I'm just trying to help. sorry if you don't like my way to do that

**Comment by FoxPotato** [comment]
I appreciate it, sorry. I didn't have enough sugar in my blood, I get grumpy. And I just had an exam i kinda f'ed up.

**Comment by DarkholmeTenk** [comment]
@Soyaro666 Message me on the forums (http://www.minecraftforum.net/members/ShaneDarkholme) and I'll give you the controls formula :)

**Comment by Soyaro666** [comment]
got the formula. wrote a new one. Not sure if Darkholme is even thinking about changing that... Let's see what will happen now...
a suggestion to go back to topic: there are buttons that load settings that lead to current position and last position. maybe those can be redone to load settings in absolute mode that bring the tardis as close as possible to its current/last position? so the Tardis would only need to save current/last coordinates instead of the whole settings. that would consume less memory (which doesn't matter while you're playing with just a few tardises of course)
